contracts/assemblyscript/erc20/assembly/index.ts

```diff
@@ -0,0 +1,139 @@
+import { u128 } from "bignum";
+import { hash } from "../node_modules/@chainsafe/as-sha256/assembly";
+
+import {
+  getCaller,
+  getScratchBuffer,
+  getStorage,
+  getValueTransferred,
+  printLine,
+  setRentAllowance,
+  setScratchBuffer,
+  setStorage
+} from "./lib";
+
+import {
+  getBalanceOrZero,
+  mergeToSha256,
+  toBytes
+} from "./lib-contract";
+
+/**
+ *  Read the Ethereum ERC20 specs https://github.com/ethereum/EIPs/blob/master/EIPS/eip-20.md
+ **/ 
+
+const ERC20_SUPPLY_STORAGE = (new Uint8Array(32)).fill(3);
+
+enum Action {
+  TotalSupply, // -> Balance
+  BalanceOf, // -> Balance
+  Transfer, // -> bool
+  TransferFrom, // -> bool
+  Approve, // -> bool
+  Allowance, // -> Balance
+  SelfEvict
```

Comment:
Do you actually need this? It is not a part of the ERC20 nor used in tests

Reply:
No, you're right, I can remove it.